### ISSUE TITLE
feat: generated tests should be final and strict 

### ISSUE THREAD

**Issue by nopenopenope** [paul-thebaud/phpunitgen-core#19]
### Subject of the issue
Right now, generated Tests are somewhat open. However, a better approach to create PHP files is to declare them final and declare strict types. 

Reasoning for that is that Tests should not be extended from, so naturally, they should be marked as `final`. Using strict declaration allows developers to more strictly create their tests, thus further hardening their tests.

Tests are created like this:
```
<?php

namespace My\Custom\Namespace;

use ...

class MyTest extends TestCase
```

A better approach would be if the classes would look like this:

```
<?php

declare(strict_types=1);

namespace My\Custom\Namespace;

use ...

final class MyTest extends TestCase
```

By doing this, eventually Tests become more strict and less "flasky". Right now, many tools for Code Style and Smell suggest to do exactly that, and personally, I work in some projects that use for example [ECS](https://github.com/symplify/easy-coding-standard) that are individually setup to complain about just that. This experience comes from huge eCommerce projects, so maybe another Developers mileage differs here. 

tl;dr: I suggest to make tests more strict and final. :)

### Your environment
* version of `phpunitgen/core`: 1
* version of `phpunitgen/console`: 1.5

**Comment by paul-thebaud** [comment]
Hi,
I agree with this suggestions, and I'll try to add the final keyword soon. However, for the declare strict, I think this should be a non default behavior with an option, because it might fail with new tests for regular users. I'll keep in touch for this.

**Comment by nopenopenope** [comment]
That sounds great, and makes lot of sense to me. Thanks for considering!

**Comment by nopenopenope** [comment]
As for #21, I will implement the change in a Magento-based contribution where I will persist my changes directly into. I think its best to have these features isolated in a Magento2-based feature, so that the existing Laravel Tests might not suffer from any implication of this.

**Comment by paul-thebaud** [comment]
Hi,
This feature can benefit to any users (Magento, Laravel or any other) so I'll try to implement it this week on my free time to save your time.
This is an independent feature and I think it should be detached from any Magento specific feature.
You're ok with this?

**Comment by nopenopenope** [comment]
Yes, I am okay with this. The change cost me around 10 minutes to make in my hacky env where I adapted the package to my needs, so I am sure it will be easy for you just as well. Go ahead, I am happy to see that change coming! :+1:

**Comment by paul-thebaud** [comment]
Hi @nopenopenope, this changes is currently in a PR, and will be released on the main branch. Since it contains breaking changes (see the PR), it will be released on a new core major release. I think we should wait for the different issue to be done in this main branch before releasing the new major release, to avoid multiple major release and avoid reducing maintenability.
